tests/components/stookwijzer/test_init.py

```diff
@@ -0,0 +1,82 @@
+"""Test the Stookwijzer init."""
+
+from unittest.mock import patch
+
+from homeassistant.components.sensor import DOMAIN as SENSOR_DOMAIN
+from homeassistant.components.stookwijzer.const import DOMAIN
+from homeassistant.components.stookwijzer.sensor import STOOKWIJZER_SENSORS
+from homeassistant.config_entries import ConfigEntryState
+from homeassistant.core import HomeAssistant
+
+from . import setup_integration
+
+from tests.test_util.aiohttp import AiohttpClientMocker
+
+
+async def test_load_unload_config_entry(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the Stookwijzer configuration entry loading and unloading."""
+    entry = await setup_integration(hass, aioclient_mock, True, True)
+
+    assert entry.runtime_data is not None
+    assert entry.state is ConfigEntryState.LOADED
+    assert entry.unique_id is not None
+
+    for description in STOOKWIJZER_SENSORS:
+        state = hass.states.get(f"{SENSOR_DOMAIN}.{DOMAIN}_{description.key}")
+        assert state.state in {"0", "2", "2.0", "code_yellow"}
+
+    await hass.config_entries.async_unload(entry.entry_id)
+    await hass.async_block_till_done()
+
+    assert entry.state is ConfigEntryState.NOT_LOADED
+
+
+async def test_transform_failure(
+    hass: HomeAssistant, aioclient_mock: AiohttpClientMocker
+) -> None:
+    """Test the Stookwijzer configuration entry loading and unloading."""
+    entry = await setup_integration(hass, aioclient_mock, False)
+
+    assert entry.state is ConfigEntryState.SETUP_ERROR
```

Comment:
Let's also use caplog to check the error in the logs

Reply:
This should have lead to an migration error, which has been fixed now.